com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/util/SpecialResourceTypesRequiredParametersUtilTest.java

```diff
@@ -0,0 +1,72 @@
+package com.sap.cloud.lm.sl.cf.core.util;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.sap.cloud.lm.sl.cf.core.cf.v1.ResourceType;
+import com.sap.cloud.lm.sl.cf.core.cf.v1.SpecialResourceTypesRequiredParameters;
+import com.sap.cloud.lm.sl.common.ContentException;
+
+class SpecialResourceTypesRequiredParametersUtilTest {
+
+    @Test
+    void getRequiredParametersForManagedService() {
+        ResourceType resourceType = ResourceType.MANAGED_SERVICE;
+        Set<String> expectedRequiredParameters = SpecialResourceTypesRequiredParameters.getRequiredParameters(resourceType);
+        Set<String> actualRequiredParameters = SpecialResourceTypesRequiredParametersUtil.getRequiredParameters(resourceType);
```

Comment:
You should only test the public methods of a class (black box testing). Private and protected methods are an implementation detail and change more regularly than the public methods, which would make their tests very fragile.